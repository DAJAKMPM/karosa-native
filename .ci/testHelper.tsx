import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
