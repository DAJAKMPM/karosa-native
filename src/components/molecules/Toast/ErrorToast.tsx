import React, { FC, ComponentProps } from "react";
import Icon from "@app/atoms/Icon";
import { IMAGE_SIZE } from "./config";
import BaseToast from "./BaseToast";

export const ErrorToast: FC<ComponentProps<typeof BaseToast>> = (props) => {
  return (
    <BaseToast
      {...props}
      iconElement={
        <Icon
          group="common"
          name="warning"
          height={IMAGE_SIZE}
          width={IMAGE_SIZE}
        />
      }
    />
  );
};
