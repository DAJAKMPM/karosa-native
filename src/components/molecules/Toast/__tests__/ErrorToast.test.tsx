import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import { noop } from "lodash";
import { ErrorToast } from "../ErrorToast";

describe("ErrorToast Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<ErrorToast onClose={noop} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
