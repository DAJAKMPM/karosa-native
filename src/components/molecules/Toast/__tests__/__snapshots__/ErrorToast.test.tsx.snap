// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorToast Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(64, 64, 64, 0.87)",
        "borderRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": "column",
        "height": 72,
        "justifyContent": "center",
        "opacity": 0.95,
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
        "width": "80%",
      }
    }
  >
    <View>
      <SvgMock
        height={32}
        width={32}
      />
    </View>
    <View />
  </View>
</View>
`;
